data/index.js

```diff
@@ -139,7 +155,18 @@ export function select( reducerKey ) {
 		return select( reducerKey )[ selectorKey ]( ...args );
 	}
 
-	return selectors[ reducerKey ];
+	const createResolver = ( selector, key ) => ( ...args ) => {
+		const result = selector( ...args );
+		const resolver = resolvers[ reducerKey ] && resolvers[ reducerKey ][ key ];
+		if ( ! resolver ) {
+			return result;
+		}
+		resolver.fulfill( ...args );
+
+		return result;
```

Comment:
If both logic paths `return result;`, maybe it'd be simple to move `fulfill()` into the condition above as negated?

```js
if ( resolver ) {
	resolver.fulfill( ...args );
}

return result;
```